src/layer/x86/x86_usability.h

```diff
@@ -119,14 +119,35 @@ static inline __m128i float2int8_sse(const __m128& _v0, const __m128& _v1, const
 
 #if __AVX__
 #include <immintrin.h>
+#ifndef __AVX2__
+static inline __m256 _mm256_comp_fmadd_ps(__m256 _a, const __m256 _b, const __m256 _c)
+{
+    return _mm256_add_ps(_mm256_mul_ps(_a, _b), _c);
+}
+static inline __m128 _mm_comp_fmadd_ps(__m128 _a, const __m128 _b, const __m128 _c)
+{
+    return _mm_add_ps(_mm_mul_ps(_a, _b), _c);
+}
+#else
+static inline __m128 _mm_comp_fmadd_ps(__m128 _a, const __m128 _b, const __m128 _c)
+{
+    return _mm_fmadd_ps(_a, _b, _c);
+}
+static inline __m256 _mm256_comp_fmadd_ps(__m256 _a, const __m256 _b, const __m256 _c)
+{
+    return _mm256_fmadd_ps(_a, _b, _c);
+}
+#endif
+#if __AVX2__
```

Comment:
I tried that, but I get this error:
```
In file included from /Users/tverhulsdonck/Timen/ncnn/build/src/layer/x86/batchnorm_x86_avx.cpp:23:
/Users/tverhulsdonck/Timen/ncnn/src/layer/x86/x86_usability.h:123:22: error: redefinition of '_mm256_fmadd_ps'
static inline __m256 _mm256_fmadd_ps(__m256 _a, const __m256 _b, const __m256 _c)
                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.3/include/fmaintrin.h:142:1: note: previous definition is here
_mm256_fmadd_ps(__m256 __A, __m256 __B, __m256 __C)
```

Reply:
> I tried that, but I get this error:
> 
> ```
> ncnn/build/src/layer/x86/batchnorm_x86_avx.cpp:23:
> Timen/ncnn/src/layer/x86/x86_usability.h:123:22: error: redefinition of '_mm256_fmadd_ps'
> static inline __m256 _mm256_fmadd_ps(__m256 _a, const __m256 _b, const __m256 _c)
>                      ^
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/11.0.3/include/fmaintrin.h:142:1: note: previous definition is here
> _mm256_fmadd_ps(__m256 __A, __m256 __B, __m256 __C)
> ```

so, avx defined and avx2 not defined != fma not defined

```cpp
#if __FMA__
// use system fma
#else
// our fake fma
#endif
```

Reply:
Which makes sense as this is the define in that file:
```
static __inline__ __m256 __DEFAULT_FN_ATTRS256
_mm256_fmadd_ps(__m256 __A, __m256 __B, __m256 __C)
{
  return (__m256)__builtin_ia32_vfmaddps256((__v8sf)__A, (__v8sf)__B, (__v8sf)__C);
}
```

Reply:
Can you give a concrete example? My other option is to replace the fmadd call everywhere in each layer with a switched one based on compiler directives. This would be a bit more of a mess.

Reply:
> Which makes sense as this is the define in that file:
> 
> ```
> static __inline__ __m256 __DEFAULT_FN_ATTRS256
> _mm256_fmadd_ps(__m256 __A, __m256 __B, __m256 __C)
> {
>   return (__m256)__builtin_ia32_vfmaddps256((__v8sf)__A, (__v8sf)__B, (__v8sf)__C);
> }
> ```

Indeed, after testing on godbolt, the compiler sucks.
we have to live with the comp function anyway